config/settings.txt
bool Window.RequireForeground=True
bool Window.ShowIngameMenu=True
bool ClientHacks=True
bool ClientHacks.Maphack=False
bool ClientHacks.Zoomhack=False
bool ClientHacks.Fullbright=False
bool ClientHacks.Particles=False
bool PreloadAlert=True
bool Healthbars=True
bool Healthbars.ShowES=True
bool Healthbars.ShowIncrements=True
bool Healthbars.ShowInTown=True
bool MinimapIcons=True
bool MinimapIcons.Monsters=True
bool MinimapIcons.Strongboxes=True
bool MinimapIcons.Chests=True
bool MinimapIcons.Minions=True
bool MinimapIcons.AlertedItems=True
bool MinimapIcons.Masters=True
bool ItemAlert=True
bool ItemAlert.Rares=True
bool ItemAlert.Uniques=True
bool ItemAlert.Maps=True
bool ItemAlert.Currency=True
bool ItemAlert.SkillGems=True
bool ItemAlert.QualitySkillGems=True
bool ItemAlert.RGB=True
bool ItemAlert.PlaySound=True
bool ItemAlert.ShowText=True
bool ItemAlert.Crafting=True
bool Tooltip=True
bool Tooltip.ShowItemLevel=True
bool MonsterTracker=True
bool MonsterTracker.ShowText=True
bool MonsterTracker.PlaySound=True
bool Healthbars.Players=False
bool Healthbars.Minions=True
bool Healthbars.Enemies=True
bool Healthbars.Enemies.Normal=True
bool Healthbars.Enemies.Magic=True
bool Healthbars.Enemies.Rare=True
bool Healthbars.Enemies.Unique=True
bool XphDisplay=True
bool ExitWithGame=True
bool DpsDisplay=True
bool Healthbars.Enemies.Normal.PrintPercents=False
bool Healthbars.Enemies.Normal.PrintHealthText=False
bool Healthbars.Enemies.Magic.PrintPercents=False
bool Healthbars.Enemies.Magic.PrintHealthText=True
bool Healthbars.Enemies.Rare.PrintPercents=False
bool Healthbars.Enemies.Rare.PrintHealthText=True
bool Healthbars.Enemies.Unique.PrintPercents=False
bool Healthbars.Enemies.Unique.PrintHealthText=True
bool Tooltip.ShowItemMods=True
int Menu.Size=25
int Menu.Length=50
int Menu.PositionWidth=0
int Menu.PositionHeight=100
int PreloadAlert.FontSize=12
int PreloadAlert.BgAlpha=180
int XphDisplay.FontSize=12
int XphDisplay.BgAlpha=180
int ItemAlert.MinLinks=5
int ItemAlert.MinSockets=6
int ItemAlert.ShowText.FontSize=15
int ItemAlert.PositionHeight=0
int ItemAlert.PositionWidth=0
int MonsterTracker.ShowText.FontSize=16
int MonsterTracker.ShowText.BgAlpha=128
int Healthbars.Players.Width=105
int Healthbars.Players.Height=25
int Healthbars.Minions.Width=105
int Healthbars.Minions.Height=25
int Healthbars.Enemies.Normal.Width=105
int Healthbars.Enemies.Normal.Height=25
int Healthbars.Enemies.Magic.Width=105
int Healthbars.Enemies.Magic.Height=25
int Healthbars.Enemies.Rare.Width=105
int Healthbars.Enemies.Rare.Height=25
int Healthbars.Enemies.Unique.Width=105
int Healthbars.Enemies.Unique.Height=25
int ItemAlert.QualitySkillGemsLevel=1
int DangerAlert.ShowText.FontSize=16
int DangerAlert.ShowText.BgAlpha=128
string Window.Name=Something in Galaxy
color Healthbars.Players.Color=FF008000
color Healthbars.Minions.Color=FF90EE90
color Healthbars.Players.Outline=0
color Healthbars.Minions.Outline=0
color Healthbars.Enemies.Normal.Outline=0
color Healthbars.Enemies.Magic.Outline=FF8888FF
color Healthbars.Enemies.Rare.Outline=FFFFFF77
color Healthbars.Enemies.Unique.Outline=FFFFA500
color Healthbars.Enemies.Normal.Color=FFFF0000
color Healthbars.Enemies.Magic.Color=FFFF0000
color Healthbars.Enemies.Rare.Color=FFFF0000
color Healthbars.Enemies.Unique.Color=FFFF0000
color Healthbars.Players.PercentTextColor=FFE6E600
color Healthbars.Minions.PercentTextColor=FFE6E600
color Healthbars.Enemies.Normal.Under10Percent=FFFFFFFF
color Healthbars.Enemies.Normal.PercentTextColor=FF66FF66
color Healthbars.Enemies.Normal.HealthTextColor=FFFFFFFF
color Healthbars.Enemies.Normal.HealthTextColorUnder10Percent=FFFFFF00
color Healthbars.Enemies.Magic.Under10Percent=FFFFFFFF
color Healthbars.Enemies.Magic.PercentTextColor=FF66FF99
color Healthbars.Enemies.Magic.HealthTextColor=FFFFFFFF
color Healthbars.Enemies.Magic.HealthTextColorUnder10Percent=FFFFFF00
color Healthbars.Enemies.Rare.Under10Percent=FFFFFFFF
color Healthbars.Enemies.Rare.PercentTextColor=FF66FF99
color Healthbars.Enemies.Rare.HealthTextColor=FFFFFFFF
color Healthbars.Enemies.Rare.HealthTextColorUnder10Percent=FFFFFF00
color Healthbars.Enemies.Unique.Under10Percent=FFFFFFFF
color Healthbars.Enemies.Unique.PercentTextColor=FF66FF99
color Healthbars.Enemies.Unique.HealthTextColor=FFFFFFFF
color Healthbars.Enemies.Unique.HealthTextColorUnder10Percent=FFFFFF00


config/new_settings.txt
Client Hacks = True
Client Hacks.Full bright = False
Client Hacks.Maphack = False
Client Hacks.Suspend Particles = False
Client Hacks.Zoomhack = False
DPS-meter = True
DPS-meter.DPS font size = 16
DPS-meter.Peak font size = 8
Health Bars = True
Health Bars.Enemies.Magic = True
Health Bars.Enemies.Magic.Color = FFFF0000
Health Bars.Enemies.Magic.Color for Percents Text = FFFFFFFF
Health Bars.Enemies.Magic.Color when under 10% = FFFFFFFF
Health Bars.Enemies.Magic.Health Text Color = FFFFFFFF
Health Bars.Enemies.Magic.Height = 25
Health Bars.Enemies.Magic.Outline = FF8888FF
Health Bars.Enemies.Magic.Show Health Value = False
Health Bars.Enemies.Magic.Show Percents = False
Health Bars.Enemies.Magic.Text Color when under 10% = FFFF0000
Health Bars.Enemies.Magic.Width = 105
Health Bars.Enemies.Normal = True
Health Bars.Enemies.Normal.Color = FFFF0000
Health Bars.Enemies.Normal.Color for Percents Text = FFFFFFFF
Health Bars.Enemies.Normal.Color when under 10% = FFFFFFFF
Health Bars.Enemies.Normal.Health Text Color = FFFFFFFF
Health Bars.Enemies.Normal.Height = 25
Health Bars.Enemies.Normal.Outline = 0
Health Bars.Enemies.Normal.Show Health Value = False
Health Bars.Enemies.Normal.Show Percents = False
Health Bars.Enemies.Normal.Text Color when under 10% = FFFF0000
Health Bars.Enemies.Normal.Width = 105
Health Bars.Enemies.Rare = True
Health Bars.Enemies.Rare.Color = FFFF0000
Health Bars.Enemies.Rare.Color for Percents Text = FFFFFFFF
Health Bars.Enemies.Rare.Color when under 10% = FFFFFFFF
Health Bars.Enemies.Rare.Health Text Color = FFFFFFFF
Health Bars.Enemies.Rare.Height = 25
Health Bars.Enemies.Rare.Outline = FFFFFF77
Health Bars.Enemies.Rare.Show Health Value = True
Health Bars.Enemies.Rare.Show Percents = False
Health Bars.Enemies.Rare.Text Color when under 10% = FFFF0000
Health Bars.Enemies.Rare.Width = 105
Health Bars.Enemies.Unique = True
Health Bars.Enemies.Unique.Color = FFFF0000
Health Bars.Enemies.Unique.Color for Percents Text = FFFFFFFF
Health Bars.Enemies.Unique.Color when under 10% = FFFFFFFF
Health Bars.Enemies.Unique.Health Text Color = FFFFFFFF
Health Bars.Enemies.Unique.Height = 25
Health Bars.Enemies.Unique.Outline = FFAF6025
Health Bars.Enemies.Unique.Show Health Value = True
Health Bars.Enemies.Unique.Show Percents = False
Health Bars.Enemies.Unique.Text Color when under 10% = FFFF0000
Health Bars.Enemies.Unique.Width = 105
Health Bars.Minions = True
Health Bars.Minions.Color = FF90EE90
Health Bars.Minions.Color for Percents Text = FFFFFFFF
Health Bars.Minions.Color when under 10% = FFFFFFFF
Health Bars.Minions.Health Text Color = FFFFFFFF
Health Bars.Minions.Height = 25
Health Bars.Minions.Outline = 0
Health Bars.Minions.Show Health Value = True
Health Bars.Minions.Show Percents = True
Health Bars.Minions.Text Color when under 10% = FFFF0000
Health Bars.Minions.Width = 105
Health Bars.Players = False
Health Bars.Players.Color = FF008000
Health Bars.Players.Color for Percents Text = FFFFFFFF
Health Bars.Players.Color when under 10% = FFFFFFFF
Health Bars.Players.Health Text Color = FFFFFFFF
Health Bars.Players.Height = 25
Health Bars.Players.Outline = 0
Health Bars.Players.Show Health Value = True
Health Bars.Players.Show Percents = True
Health Bars.Players.Text Color when under 10% = FFFF0000
Health Bars.Players.Width = 105
Health Bars.Show ES = True
Health Bars.Show In Town = True
Health Bars.Show Increments = True
Item Alert = True
Item Alert.Crafting Bases = True
Item Alert.Currency = True
Item Alert.Font Size = 14
Item Alert.Maps = True
Item Alert.Play Sound = True
Item Alert.Rares = True
Item Alert.Show Text = True
Item Alert.Skill Gems = True
Item Alert.Skill Gems.Minimum Quality = 0
Item Alert.Sockets.Chrome Link = True
Item Alert.Sockets.Minimum Links = 5
Item Alert.Sockets.Minimum Sockets = 6
Item Alert.Uniques = True
Item Tooltips = True
Item Tooltips.Item Level = True
Item Tooltips.Mod Rolls = True
Map Icons = True
Map Icons. Show on.Large map = True
Map Icons. Show on.Minimap = True
Map Icons.Chests = True
Map Icons.Masters = True
Map Icons.Monsters = True
Map Icons.Monsters.Magic = True
Map Icons.Monsters.Normal = True
Map Icons.Monsters.Rare = True
Map Icons.Monsters.Unique = True
Map Icons.Strongboxes = True
Map Icons.Valuable items = True
Menu = True
Menu.Anchor Height = 25
Menu.Anchor Width = 50
Menu.Item Height = 25
Menu.Item Width = 200
Menu.PositionHeight = 100
Menu.PositionWidth = 0
Monster Alerts = True
Monster Alerts.Bg Alpha = 128
Monster Alerts.Font Size = 16
Monster Alerts.Play Sound = True
Monster Alerts.Show Text = True
PreloadAlert = True
PreloadAlert.BgAlpha = 160
PreloadAlert.FontSize = 12
Window.Name = ExileHUD
Window.RequireForeground = True
Window.ShowIngameMenu = True
XP Meter = True
XP Meter.Bg Opacity = 160
XP Meter.Clock = True
XP Meter.Current map name = True
XP Meter.Font Size = 12


src/Hud/Settings.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Globalization;
using System.IO;
using System.Text;
using System.Windows.Forms;

namespace PoeHUD.Hud
{
	public class Settings
	{
		private static Dictionary<string, bool> bools;
		private static Dictionary<string, int> ints;
		private static Dictionary<string, float> floats;
		private static Dictionary<string, string> strings;
		private static Dictionary<string, Color> colors;
		public static bool GetBool(string setting)
		{
			return bools.ContainsKey(setting) && bools[setting];
		}
		public static int GetInt(string setting)
		{
			if (ints.ContainsKey(setting))
			{
				return ints[setting];
			}
			return 0;
		}
		public static float GetFloat(string setting)
		{
			if (floats.ContainsKey(setting))
			{
				return floats[setting];
			}
			return 0f;
		}
		public static string GetString(string setting)
		{
			if (strings.ContainsKey(setting))
			{
				return strings[setting];
			}
			return "";
		}
		public static Color GetColor(string setting)
		{
			if (colors.ContainsKey(setting))
			{
				return colors[setting];
			}
			return Color.White;
		}
		public static void SetBool(string setting, bool value)
		{
			if (bools.ContainsKey(setting))
			{
				bools[setting] = value;
			}
			else
			{
				bools.Add(setting, value);
			}
			SaveSettings();
		}
		public static void SetInt(string setting, int value)
		{
			if (ints.ContainsKey(setting))
			{
				ints[setting] = value;
			}
			else
			{
				ints.Add(setting, value);
			}
			SaveSettings();
		}
		public static void SetColor(string setting, Color color)
		{
			if (colors.ContainsKey(setting))
			{
				colors[setting] = color;
			}
			else
			{
				colors.Add(setting, color);
			}
			SaveSettings();
		}
		public static void SetString(string setting, string value)
		{
			if (strings.ContainsKey(setting))
			{
				strings[setting] = value;
			}
			else
			{
				strings.Add(setting, value);
			}
			SaveSettings();
		}
		private static void SaveSettings()
		{
			StringBuilder stringBuilder = new StringBuilder();
			foreach (KeyValuePair<string, bool> current in bools)
			{
				stringBuilder.AppendLine(string.Concat(new object[] { "bool ", current.Key, "=", current.Value }));
			}
			foreach (KeyValuePair<string, int> current2 in ints)
			{
				stringBuilder.AppendLine(string.Concat(new object[] { "int ", current2.Key, "=", current2.Value }));
			}
			foreach (KeyValuePair<string, float> current3 in floats)
			{
				stringBuilder.AppendLine(string.Concat(new object[] { "float ", current3.Key, "=", current3.Value }));
			}
			foreach (KeyValuePair<string, string> current4 in strings)
			{
				stringBuilder.AppendLine("string " + current4.Key + "=" + current4.Value);
			}
			foreach (KeyValuePair<string, Color> current5 in colors)
			{
				stringBuilder.AppendLine("color " + current5.Key + "=" + current5.Value.ToArgb().ToString("X"));
			}
			File.WriteAllText("config/settings.txt", stringBuilder.ToString());
		}
		public static bool LoadSettings()
		{
			bools = new Dictionary<string, bool>();
			ints = new Dictionary<string, int>();
			floats = new Dictionary<string, float>();
			strings = new Dictionary<string, string>();
			colors = new Dictionary<string, Color>();
			SetDefault();
			if (!File.Exists("config/settings.txt"))
			{
				MessageBox.Show("config/settings.txt does not exist!");
				return false;
			}
			string[] allLines = File.ReadAllLines("config/settings.txt");
			foreach (string text in allLines)
			{
				try
				{
					string text2 = text.Trim().Replace("\n", "").Replace("\r", "");
					if (string.IsNullOrWhiteSpace(text2)) continue;


					string[] allWords = text2.Split(new []{' '}, 2, StringSplitOptions.RemoveEmptyEntries);
					if (allWords.Length != 2)
					{
						Console.WriteLine("Invalid settings line: " + text2);
						continue;
					}

					string firstWord = allWords[0];
					string[] kv = allWords[1].Split(new []{'='}, 2, StringSplitOptions.RemoveEmptyEntries);
					if (kv.Length < 2)
					{
						Console.WriteLine("Invalid settings line: " + text2);
					}
					else
					{
						string key = kv[0].Trim();
						string value = kv[1].Trim();
						switch (firstWord.ToLowerInvariant())
						{
							case "bool":
								bools[key] = bool.Parse(value);
								break;
							case "int":
								ints[key] = int.Parse(value);
								break;
							case "float":
								floats[key] = float.Parse(value);
								break;
							case "string":
								strings[key] = value;
								break;
							case "color":
								colors[key] = Color.FromArgb(int.Parse(value, NumberStyles.HexNumber));
								break;
						}
					}
				}
				catch (Exception)
				{
					MessageBox.Show("Error in settings line: " + text);
					return false;
				}
			}
			return true;
		}
		private static void SetDefault()
		{
			ints.Add("Menu.Size", 25);
			ints.Add("Menu.Length", 50);
			ints.Add("Menu.PositionWidth", 0);
			ints.Add("Menu.PositionHeight", 100);
			bools.Add("Window.RequireForeground", false);
			bools.Add("Window.ShowIngameMenu", true);
			bools.Add("ClientHacks", false);
			bools.Add("ClientHacks.Maphack", false);
			bools.Add("ClientHacks.Zoomhack", false);
			bools.Add("ClientHacks.Fullbright", false);
			bools.Add("ClientHacks.Particles", false);
			bools.Add("PreloadAlert", true);
			bools.Add("Healthbars", true);
			bools.Add("Healthbars.ShowES", false);
			bools.Add("Healthbars.ShowIncrements", true);
			bools.Add("Healthbars.ShowInTown", true);
			bools.Add("MinimapIcons", true);
			bools.Add("MinimapIcons.Monsters", true);
			bools.Add("MinimapIcons.Strongboxes", true);
			bools.Add("MinimapIcons.Chests", true);
			bools.Add("MinimapIcons.Minions", true);
			bools.Add("MinimapIcons.AlertedItems", true);
            bools.Add("MinimapIcons.Masters", true);
			bools.Add("ItemAlert", true);
			bools.Add("ItemAlert.Rares", true);
			bools.Add("ItemAlert.Uniques", true);
			bools.Add("ItemAlert.Maps", true);
			bools.Add("ItemAlert.Currency", true);
			bools.Add("ItemAlert.SkillGems", true);
			bools.Add("ItemAlert.QualitySkillGems", true);
			bools.Add("ItemAlert.RGB", false);
			bools.Add("ItemAlert.PlaySound", true);
			bools.Add("ItemAlert.ShowText", true);
			bools.Add("ItemAlert.Crafting", true);
			bools.Add("Tooltip", true);
			bools.Add("Tooltip.ShowItemLevel", true);
			bools.Add("MonsterTracker", true);
			bools.Add("MonsterTracker.ShowText", true);
			bools.Add("MonsterTracker.PlaySound", true);
			bools.Add("Healthbars.Players", true);
			bools.Add("Healthbars.Minions", true);
			bools.Add("Healthbars.Enemies", true);
			bools.Add("Healthbars.Enemies.Normal", true);
			bools.Add("Healthbars.Enemies.Magic", true);
			bools.Add("Healthbars.Enemies.Rare", true);
			bools.Add("Healthbars.Enemies.Unique", true);
			bools.Add("XphDisplay", true);
			bools.Add("ExitWithGame", true);
			bools.Add("DpsDisplay", true);
			ints.Add("PreloadAlert.FontSize", 12);
			ints.Add("PreloadAlert.BgAlpha", 180);
			ints.Add("XphDisplay.FontSize", 12);
			ints.Add("XphDisplay.BgAlpha", 180);
			ints.Add("ItemAlert.MinLinks", 5);
			ints.Add("ItemAlert.MinSockets", 6);
			ints.Add("ItemAlert.ShowText.FontSize", 16);
			ints.Add("ItemAlert.PositionHeight", 200);
			ints.Add("ItemAlert.PositionWidth", 0);
			ints.Add("MonsterTracker.ShowText.FontSize", 16);
			ints.Add("MonsterTracker.ShowText.BgAlpha", 128);
			ints.Add("Healthbars.Players.Width", 105);
			ints.Add("Healthbars.Players.Height", 25);
			ints.Add("Healthbars.Minions.Width", 105);
			ints.Add("Healthbars.Minions.Height", 25);
			ints.Add("Healthbars.Enemies.Normal.Width", 105);
			ints.Add("Healthbars.Enemies.Normal.Height", 25);
			ints.Add("Healthbars.Enemies.Magic.Width", 105);
			ints.Add("Healthbars.Enemies.Magic.Height", 25);
			ints.Add("Healthbars.Enemies.Rare.Width", 105);
			ints.Add("Healthbars.Enemies.Rare.Height", 25);
			ints.Add("Healthbars.Enemies.Unique.Width", 105);
			ints.Add("Healthbars.Enemies.Unique.Height", 25);
			strings.Add("Window.Name", "PoeHUD");
			colors.Add("Healthbars.Players.Color", Color.FromArgb(255, 0, 128, 0));
			colors.Add("Healthbars.Minions.Color", Color.FromArgb(255, 144, 238, 144));
			colors.Add("Healthbars.Players.Outline", Color.FromArgb(0));
			colors.Add("Healthbars.Minions.Outline", Color.FromArgb(0));
			colors.Add("Healthbars.Enemies.Normal.Outline", Color.FromArgb(0));
			colors.Add("Healthbars.Enemies.Magic.Outline", Color.FromArgb(255, 136, 136, 255));
			colors.Add("Healthbars.Enemies.Rare.Outline", Color.FromArgb(255, 255, 255, 119));
			colors.Add("Healthbars.Enemies.Unique.Outline", Color.FromArgb(255, 175, 96, 37));
			colors.Add("Healthbars.Enemies.Normal.Color", Color.FromArgb(255, 255, 0, 0));
			colors.Add("Healthbars.Enemies.Magic.Color", Color.FromArgb(255, 255, 0, 0));
			colors.Add("Healthbars.Enemies.Rare.Color", Color.FromArgb(255, 255, 0, 0));
			colors.Add("Healthbars.Enemies.Unique.Color", Color.FromArgb(255, 255, 0, 0));
		}
	}
}
